<?xml version="1.0"?>
<launch>
	<!-- Input arguments -->
	<arg name="argID"               default="0"/>
	<arg name="argNodeRate"         default="100.0"/>
	<arg name="argNamespace"        default="heifu$(arg argID)"/>
	<arg name="argVehicle"          default="$(arg argNamespace)"/>
	<arg name="argVoxelSize"        default="0.5"/>
	<arg name="argErodeThreshold"   default="0.0"/>
	<arg name="argDensityThreshold" default="20"/>
	<arg name="argPointCloudTopic"  default="/pointcloud/depth/color/points"/>
	<arg name="argCloudTargetFrame" default="/odom"/>
	<arg name="argCloudSourceFrame" default="/camera_link"/>

	<!-- Launch mavros node -->
	<node name="gpu_voxels_ros" type="gpu_voxels_ros_node" pkg="gpu_voxels_ros" output="screen" required="true" ns="$(arg argNamespace)">
		<param name="paramNodeRate"         type="double" value="$(arg argNodeRate)"		/>
		<param name="paramVoxelSize"         type="double" value="$(arg argVoxelSize)"		/>
		<param name="paramErodeThreshold"         type="double" value="$(arg argErodeThreshold)"		/>

		<param name="paramDensityThreshold" type="int" value="$(arg argDensityThreshold)" />

		<param name="paramPointCloudTopic" type="string" value="$(arg argPointCloudTopic)" />
		<param name="paramCloudTargetFrame" type="string" value="$(arg argCloudTargetFrame)" />
		<param name="paramCloudSourceFrame" type="string" value="$(arg argCloudSourceFrame)" />

		<rosparam param="paramMapSize">[512, 512, 256]</rosparam>
	</node>
</launch>
